app/controllers/concerns/foreman/controller/registration.rb

```diff
@@ -30,7 +30,8 @@ def global_registration_vars
                    operatingsystem: operatingsystem,
                    url_host: registration_url.host,
                    registration_url: registration_url,
-                    })
+                   setup_insights: ActiveRecord::Type::Boolean.new.deserialize(params['setup_insights']),
```

Comment:
We should only set the value in case it differs from the current configuration of the newly registered host. Otherwise this should be `nil` and the template should not add `--data setup_insights=true|false` at all.

I believe we need to build the host object first. Something like

```ruby
current_insights_value = Host.new(organization: organization, location: location, hostgroup: host_group, operatingsystem: operatingsystem).params['setup_insights']
```

then setup_insights should be something like

```ruby
desired_insights_value = ActiveRecord::Type::Boolean.new.deserialize(params['setup_insights']) 
...
setup_insights: desired_insights_value != current_insights_value ? desired_insights_value : nil
```

and the in the global template <%= "--data \"setup_insights=#{@setup_insights}\"" unless @setup_insights.nil? -%>